File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <stdbool.h>
#include <math.h>

int main(void)
{
    int coins = 0;
    float changeowed = 0;
    do
    {
        changeowed = get_float("Change owed:");
    }
    while (changeowed < 0);
    
    changeowed = round(changeowed * 100.00);
    float qter = 25.00;
    float dme = 10.00;
    float nk = 5.00;
    float pn = 1.00;
    
    bool owechange = true;
    
    while (changeowed > 0)
    {
        if (qter <= changeowed)
        {
            
            changeowed -= qter;
            coins += 1;
            
        }
        else if (dme <= changeowed)
        {
            changeowed -= dme;
            coins += 1;
            
        }
        else if (nk <= changeowed)
        {
            changeowed -= nk;
            coins += 1;
            
        }
        else
        {   
            changeowed -= pn;
            coins += 1;
            
        }
        
    }
    
    owechange = false;
    printf("%i", coins);
    
}

File: pset6/cash/cash.py
from cs50 import get_float


def main():
    changeowed = 0.00
    
    # ask user for input
    while changeowed <= 0.00:
        changeowed = get_float("Change owed: ")

    changeowed = round(changeowed * 100.00)
    
    # initialize needed variables
    qter = 25.00
    dme = 10.00
    nk = 5.00
    pn = 1.00
    coins = 0

    owechange = True

    # keep looping and adding the greatest possible coin until 0
    while changeowed > 0:
        if qter <= changeowed:
            changeowed -= qter
            coins += 1
        elif dme <= changeowed:
            changeowed -= dme
            coins += 1
        elif nk <= changeowed:
            changeowed -= nk
            coins += 1
        else:
            changeowed -= pn
            coins += 1
    
    # print out the number of coins
    owechange = False
    print(coins)


# call main
if __name__ == "__main__":
    main()